pulp_rpm/tests/functional/api/test_distribution_tree.py

```diff
@@ -111,6 +111,69 @@ def test_dist_tree_copy_as_content(self):
         self.assertEqual(len(get_content(repo_copy)[PULP_TYPE_DISTRIBUTION_TREE]), 1)
 
 
+class SkipDistributionTreeTest(PulpTestCase):
+    """Test Distribution Trees."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.client = gen_rpm_client()
+        cls.dist_tree_api = ContentDistributionTreesApi(cls.client)
+        cls.packages_api = ContentPackagesApi(cls.client)
+        cls.repo_api = RepositoriesRpmApi(cls.client)
+        cls.remote_api = RemotesRpmApi(cls.client)
+        delete_orphans()
+
+    def do_test(self, repository=None, remote=None, skip_treeinfo=False):
+        """Sync a repository.
+
+        Args:
+            repository (pulp_rpm.app.models.repository.RpmRepository):
+                object of RPM repository
+            remote (pulp_rpm.app.models.repository.RpmRemote):
+                object of RPM Remote
+            skip_treeinfo(bool):
+                True if we should skip treeinfo in sync, False otherwise
+        Returns (pulpcore.app.models.task.Task):
+            completed sync-task
+        """
+        if repository:
+            repo = self.repo_api.read(repository.pulp_href)
+        else:
+            repo = self.repo_api.create(gen_repo())
+            self.assertEqual(repo.latest_version_href, f"{repo.pulp_href}versions/0/")
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+
+        if not remote:
+            body = gen_rpm_remote()
+            remote = self.remote_api.create(body)
+        else:
+            remote = self.remote_api.read(remote.pulp_href)
+        if skip_treeinfo:
+            repository_sync_data = RpmRepositorySyncURL(
+                remote=remote.pulp_href, skip_types=["treeinfo"]
+            )
+        else:
+            repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        return monitor_task(sync_response.task)
+
+    def test_skip_treeinfo(self):
```

Comment:
Looks like this test relies on behavior that Isn't True in 3.17; we'll need to figure out a different way to test, or just skip this in the backport :(

Reply:
Just to  close the loop - the "fix ks-repos by giving them unique names" PR depended on a 3.18 feature as written, and would have needed to be hand-coded for 3.17, and was therefore not backported. The test here relies on the behavior from that change. As a result, we're not going to backport this to 3.17 either.